scripts/generate_enhanced_test_data.py

```
@@ -0,0 +1,1138 @@
+            subprocess.run(["git", "push", "-u", "origin", "main"], cwd=target_dir, check=True)
+        else:
+            print("Remote main branch is empty. Using force-push.")
+            subprocess.run(["git", "push", "-u", "-f", "origin", "main"], cwd=target_dir, check=True)
+            
+        print("✅ Successfully pushed code to GitHub")
+    except subprocess.CalledProcessError as e:
+        print(f"❌ Git operation failed: {e}")
+        raise
+
+
+def create_issues(repo_owner: str, repo_name: str, token: str, cluster_types: List[str] = None, noise_ratio: float = 0.3):
```

Ruff (0.14.8)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)